lib/Controller/DisplayGroup.php

```diff
@@ -2869,6 +2873,120 @@ public function triggerWebhook(Request $request, Response $response, $id)
+     *        type="integer",
+     *        required=false
+     *     ),
+     *    @SWG\Parameter(
+     *        name="criteriaUpdates",
+     *        in="body",
+     *        description="The criteria updates to send to the Player",
+     *        required=true,
+     *        @SWG\Schema(
+     *            type="array",
+     *            @SWG\Items(
+     *                type="object",
+     *                @SWG\Property(property="metric", type="string"),
+     *                @SWG\Property(property="value", type="string"),
+     *                @SWG\Property(property="ttl", type="integer")
+     *            )
+     *        )
+     *     ),
+     *    @SWG\Response(
+     *        response=204,
+     *        description="Successful operation"
+     *    ),
+     *    @SWG\Response(
+     *        response=400,
+     *        description="Invalid criteria format"
+     *    )
+     *   )
+     *
+     * @param Request $request
+     * @param Response $response
+     * @param int|null $displayGroupId
+     * @return ResponseInterface|Response
+     * @throws ControllerNotImplemented
+     * @throws GeneralException
+     * @throws NotFoundException
+     * @throws PlayerActionException
+     */
+    public function pushCriteriaUpdate(Request $request, Response $response, int $displayGroupId = null): Response|ResponseInterface
+    {
+        $sanitizedParams = $this->getSanitizer($request->getParams());
+
+        // Get criteria updates
+        $criteriaUpdates = $sanitizedParams->getArray('criteriaUpdates');
+
+        // ensure criteria updates exists
+        if (empty($criteriaUpdates)) {
+            throw new InvalidArgumentException(__('No criteria found.'), 'criteriaUpdates');
+        }
+
+        // Initialize array to hold sanitized criteria updates
+        $sanitizedCriteriaUpdates = [];
+
+        // Loop through each criterion and sanitize the input
+        foreach ($criteriaUpdates as $criteria) {
+            $criteriaSanitizer = $this->getSanitizer($criteria);
+
+            // Sanitize and retrieve the metric, value, and ttl
+            $metric = $criteriaSanitizer->getString('metric');
+            $value = $criteriaSanitizer->getString('value');
+            $ttl = $criteriaSanitizer->getInt('ttl');
+
+            // Ensure each criterion has metric, value, and ttl
+            if (empty($metric) || empty($value) || empty($ttl)) {
+                // Throw an exception if any of the required fields are missing or empty
+                throw new PlayerActionException(
+                    __('Invalid criteria format. Metric, value, and ttl must all be present and not empty.')
+                );
+            }
+
+            // Add sanitized criteria
+            $sanitizedCriteriaUpdates[] = [
+                'metric' => $metric,
+                'value' => $value,
+                'ttl' => $ttl
+            ];
+        }
+
+        if ($displayGroupId !== null) {
```

Comment:
Now that I see this, it does worry me that this could potentially be 1000s of displays. Perhaps it would be more sensible to make that parameter not optional after all?

Reply:
I think your right now that I think about it. Then, I'll make it a required parameter instead of it being optional